tests/contenttypes_tests/test_fields.py

```diff
@@ -15,13 +15,16 @@ def test_str(self):
         class Model(models.Model):
             field = GenericForeignKey()
 
-        self.assertEqual(str(Model.field), "contenttypes_tests.Model.field")
+        field = Model._meta.get_field("field")
+
+        self.assertEqual(str(field), "contenttypes_tests.Model.field")
 
     def test_get_content_type_no_arguments(self):
+        field = Answer._meta.get_field("question")
         with self.assertRaisesMessage(
             Exception, "Impossible arguments to GFK.get_content_type!"
         ):
-            Answer.question.get_content_type()
+            field.get_content_type()
```

Comment:
This usage is [documented](https://docs.djangoproject.com/en/5.2/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey), so don't we need a deprecation path? I'm willing to hustle it into 6.0 before the freeze, but we have to act fast.

Reply:
Also, while here, Simon had a request for a detailed commit message in https://github.com/django/django/pull/17554#discussion_r2191302268.